server/src/transport/ws.rs

```diff
@@ -416,9 +430,11 @@ pub(crate) async fn background_task<L: Logger>(
 					})
 					.await;
 
-					tx_log_from_str(&response.result, max_log_length);
-					logger.on_response(&response.result, request_start, TransportProtocol::WebSocket);
-					let _ = sink.send_raw(response.result);
+					if let Some(response) = response {
```

Comment:
Ok, so this basically catches the case when every request is a notification; gotcha!